src/main/java/org/traccar/api/resource/PermissionsResource.java

```diff
@@ -55,30 +61,61 @@ private void checkPermission(Permission permission, boolean link) {
                 permission.getPropertyClass(), getUserId(), permission.getPropertyId());
     }
 
+    private void refreshPermissions(List<LinkedHashMap<String, Long>> entities) {
+        Set<String> ownerProperty = new HashSet<>();
+        entities.stream()
+            .map(e -> {
+                try {
+                    return new Permission(e);
+                } catch (ClassNotFoundException classNotFoundException) {
+                    classNotFoundException.printStackTrace();
+                    return null;
+                }
+            })
+            .filter(Objects::nonNull)
+            .filter(p -> ownerProperty.add(p.getOwnerClass().getName() + p.getPropertyClass().getName()))
```

Comment:
There has to be a better way of doing this based on the original `LinkedHashMap`.